design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,312 @@
+|--------------------------------------------------------------------|-------------------------------------------|-------------------------------------------------------------------------------------|
+|     api                                                            |     authn-jwt/                            |     authn-gcp/                                                                      |
+|     Mandatory base claims to validate                              |     Iss, exp, nbf (Configurable)          |     iss, exp, aud, iat, nbf (Constants)                                           |
+|     Optional base claims to validate (if exists in the token)      |     aud,sub,iat (Configurable)            |     -                                                                               |
+|     provider-uri                                                   |     Configurable   (list?)                |     Constant                                                                        |
+|     iss value                                                      |     Configurable   (list?)                |     Constant                                                                        |
+|     aud value                                                      |     Configurable   (list?)                |     Constant                                                                        |
+|     sub value                                                      |     Configurable   (list?)                |     -                                                                               |
+|     Host   annotation prefix                                       |     authn-jwt                             |     authn-gcp                                                                       |
+|     Permitted annotation keys                                      |     any                                   |     instance-name OR project-id OR   service-account-id OR service-account-email    |
+|     Mapping host annotation logic                                  |     -                                     |     Map each above key   to his location in the google JWT token                    |
+
+The following diagram represents JWT authenticator with a lot of small responsibility units, according to the above table and past experience, 
+in order to have the flexibility to support future vendors that uses JWT token for their authentication process, like google, GCP
+
+**Note:** in this feature we will support only GCP
+![Authn gcp class diagram](authn-gcp-class-diagram.png)
+
+GCP & JWT will have 3 dependencies which will change their behavior by dependency injection  
+
+1. **Fetch Authenticator configuration**
+
+    1.1 **Fetch provider uri**
+    
+    * GCP implementation - return an hardcoded
+    * Authenticator policy implementation - fetch from provider uri value in the policy, this logic is relevant for: OIDC, Azure, JWT
```

Comment:
where is it going to be kept in policy ?

Reply:
Yes